### PR TITLE
repair(scripts/Quest): catch the wild wolpertinger!

### PR DESCRIPTION
**Pull request by grimgravy**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/pull/17333 (merged)

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Events/brewfest.cpp [hunk 1]
```diff
@@ -2057,6 +2064,52 @@ class spell_direbrew_disarm : public AuraScript
...
+    }
+
+    void SpellHit(Unit* caster, SpellInfo const* spell) override
+    {
+        if (spell->Id == SPELL_WOLPERTINGER_NET)
+        {
+            PlayerGUID = caster->GetGUID();
+            me->SetReactState(REACT_PASSIVE);
+            me->CombatStop(false);
+            GetPlayer()->CombatStop();
```
Comment (Kitzunu) [inline src/server/scripts/Events/brewfest.cpp:2087]: is player and caster not the same unit in this case?

Can't we do caster->ToPlayer()->CombatStop();?

Comment (grimgravy) [reply src/server/scripts/Events/brewfest.cpp:2087]: you're right, I wanted to make things easier

File: data/sql/updates/pending_db_world/rev_1695559726583951400.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+--
+DELETE FROM `conditions` WHERE (`SourceTypeOrReferenceId` = 17) AND (`SourceGroup` = 0) AND (`SourceEntry` = 41621) AND (`SourceId` = 0) AND (`ElseGroup` = 0) AND (`ConditionTypeOrReference` = 31) AND (`ConditionTarget` = 1) AND (`ConditionValue1` = 3) AND (`ConditionValue2` = 23487) AND (`ConditionValue3` = 0);
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
+(17, 0, 41621, 0, 0, 31, 1, 3, 23487, 0, 0, 0, 0, '', 'Throw a net at a wild wolpertinger, which will allow you to capture it and place it in your pack.');
```
Comment (Nyeriah) [inline data/sql/updates/pending_db_world/rev_1695559726583951400.sql:4]: description should reflect what the condition requires e.g "Throw net requires XXX"

File: src/server/scripts/Events/brewfest.cpp [hunk 3]
```diff
@@ -2057,6 +2064,52 @@ class spell_direbrew_disarm : public AuraScript
     }
 };
 
+struct npc_wild_wolpertinger : public ScriptedAI
+{
+public:
+    npc_wild_wolpertinger(Creature* creature) : ScriptedAI(creature) { }
+
+    ObjectGuid PlayerGUID;
```
Comment (Kitzunu) [inline src/server/scripts/Events/brewfest.cpp:2072]: Also, shouldn't this should be private?

File: data/sql/updates/pending_db_world/rev_1695559726583951400.sql (hunk 2 above)
Comment (grimgravy) [reply data/sql/updates/pending_db_world/rev_1695559726583951400.sql:4]: This is exactly what the condition reflects

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1695559726583951400.sql:4]: What Nyeriah is getting at is the comment should be like "Spell Throw Net requires Unit Wild Wolpertinger"

**Comment by sudlud** [comment]
What's the state of this PR?
Can you please respond the the open reviews and maybe update the PR accordingly 
🙂

**Comment by sudlud** [comment]
closing this PR as the author does not responded to reviews.

Please feel free to re-open this PR and address the open reviews/comments.

**Pull request closed without merging**